### PR TITLE
Add new service to transfer assignments, removes duplicate code

### PR DESCRIPTION
**Pull request by shaunakpp**
## What

We had duplicate code for transferring assignments from one admin to other. So, in this PR I have created a service `TransferAssignmentsService` which does the transfer process and is called instead of the duplicate code.

P.S. I discovered this when working on #1864 and hence closing the original PR in favor of this.

### CODE REVIEW

File: app/services/transfer_assignments_service.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class TransferAssignmentsService
+  attr_accessor :organization, :old_user, :new_user
```
Comment (d12) [inline app/services/transfer_assignments_service.rb]: Can we add a newline after this, to be more consistent with the app?

File: app/services/transfer_assignments_service.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    @old_user = old_user
+    @new_user = new_user || @organization.users.where.not(id: old_user.id).first
+  end
+
+  def transfer
+    return false unless user_owns_any_assignments?
+    all_assignments_of_user = organization.all_assignments.select do |assignment|
+      assignment.creator_id == old_user.id
+    end
+    all_assignments_of_user.map do |assignment|
```
Comment (d12) [inline app/services/transfer_assignments_service.rb]: I'd group this all together to be a little more concise:

```ruby
organization.all_assignments.map do |assignment|
  next unless assignment.creator_id == old_user.id
  assignment.update(creator_id: new_user.id)
end
```

File: app/services/transfer_assignments_service.rb (hunk 1 above)
Comment (shaunakpp) [reply app/services/transfer_assignments_service.rb]: done 👍🏼

File: app/services/transfer_assignments_service.rb (hunk 2 above)
Comment (shaunakpp) [reply app/services/transfer_assignments_service.rb]: done 👍🏼

File: spec/services/transfer_assignments_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    end
+
+    it "is expected to transfer assignments to specified new user" do
+      assignment = create(:assignment, organization: organization, creator: old_user)
+      TransferAssignmentsService.new(organization, old_user, new_user).transfer
+      assignment.reload
+      expect(assignment.creator_id).to eq(new_user.id)
+    end
+  end
+  context "when teacher owns group assignments" do
```
Comment (d12) [inline spec/services/transfer_assignments_service_spec.rb]: Can we add a newline above here too?

File: spec/services/transfer_assignments_service_spec.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+describe TransferAssignmentsService do
+  let(:organization) { classroom_org }
+  let(:old_user) { classroom_teacher }
+  let(:new_user) { create(:user) }
+  before(:each) do
+    organization.users << new_user
+    organization.save
+  end
+  context "when teacher owns no assignments" do
```
Comment (d12) [inline spec/services/transfer_assignments_service_spec.rb]: Can we add a newline above here too?

File: spec/services/transfer_assignments_service_spec.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe TransferAssignmentsService do
+  let(:organization) { classroom_org }
+  let(:old_user) { classroom_teacher }
+  let(:new_user) { create(:user) }
+  before(:each) do
```
Comment (d12) [inline spec/services/transfer_assignments_service_spec.rb]: Can we add a newline above here too?

**Pull request merged**